plugins/snowflake/dbt/adapters/snowflake/connections.py

```diff
@@ -197,6 +199,23 @@ def _get_private_key(cls, private_key_path, private_key_passphrase):
             format=serialization.PrivateFormat.PKCS8,
             encryption_algorithm=serialization.NoEncryption())
 
+    @classmethod
+    def process_results(cls, matrix):
+        # Override for Snowflake. The datetime objects returned by
+        # snowflake-connector-python are not pickleable, so we need
+        # to replace them with sane timezones
+        fixed = []
+        for row in matrix:
+            fixed_row = []
+            for col in row:
+                if isinstance(col, datetime.datetime):
+                    col = col.astimezone(tz=pytz.UTC)
```

Comment:
I like this approach a lot: force-converting everything to UTC seems fine to me and this is the right place to do it I think (I'd rather this than have RPC be special and funky, if we can avoid it, right?)

The only suggestion I have is that maybe the base class should do the `[dict(zip(column_names, row)) for row in results]` in its `process_results`, and snowflake should just do this and then `return super(SnowflakeConnector, self).process_results(fixed)` at the end? That's more of a style thing though and I don't have strong feelings about it, converting to a dict just feels like it's definitely "processing results".